In [1]:
%matplotlib inline
import pandas as pd

In [2]:
import os

* The more analysis I do with my data files the more that I adapt my research question based on the conclusions I am forming. 

* At this point my research questions are:
    - Does the proximity of Penn's campus to West Philadelphia schools have any impact on their students?
    - What do the schools with the highest and lowest rates of graduation, enrollment, etc. have in common with each other?
    - What are there any causal factors that drive the each schools high / low score? 

* I uploaded all of my refined datasets and files that I have used throughout my analysis in order to work with more specific data in a fresh notebook.

In [3]:
pd.read_csv('../data/enrollment_demographics.csv')

School Year  SRC School ID                                  School Name  \
0      2018-2019            101                     John Bartram High School   
1      2018-2019            101                     John Bartram High School   
2      2018-2019            101                     John Bartram High School   
3      2018-2019            101                     John Bartram High School   
4      2018-2019            101                     John Bartram High School   
5      2018-2019            102                West Philadelphia High School   
6      2018-2019            102                West Philadelphia High School   
7      2018-2019            102                West Philadelphia High School   
8      2018-2019            102                West Philadelphia High School   
9      2018-2019            102                West Philadelphia High School   
10     2018-2019            103                    High School of the Future   
11     2018-2019            103                    High School of the Future   
12     2018-2019            103                    High School of the Future   
13     2018-2019            103                    High School of the Future   
14     2018-2019            103                    High School of the Future   
15     2018-2019            105  Paul Robeson High School for Human Services   
16     2018-2019            105  Paul Robeson High School for Human Services   
17     2018-2019            105  Paul Robeson High School for Human Services   
18     2018-2019            105  Paul Robeson High School for Human Services   
19     2018-2019            105  Paul Robeson High School for Human Services   
20     2018-2019            110                 William L. Sayre High School   
21     2018-2019            110                 William L. Sayre High School   
22     2018-2019            110                 William L. Sayre High School   
23     2018-2019            110                 William L. Sayre High School   
24     2018-2019            110                 William L. Sayre High School   
25     2018-2019            113                     William T. Tilden School   
26     2018-2019            113                     William T. Tilden School   
27     2018-2019            113                     William T. Tilden School   
28     2018-2019            113                     William T. Tilden School   
29     2018-2019            113                     William T. Tilden School   
...          ...            ...                                          ...   
1643   2018-2019            844                    William H. Loesche School   
1644   2018-2019            844                    William H. Loesche School   
1645   2018-2019            844                    William H. Loesche School   
1646   2018-2019            844                    William H. Loesche School   
1647   2018-2019            844                    William H. Loesche School   
1648   2018-2019            844                    William H. Loesche School   
1649   2018-2019            844                    William H. Loesche School   
1650   2018-2019            856                          The Workshop School   
1651   2018-2019            856                          The Workshop School   
1652   2018-2019            856                          The Workshop School   
1653   2018-2019            856                          The Workshop School   
1654   2018-2019            856                          The Workshop School   
1655   2018-2019            877            John Hancock Demonstration School   
1656   2018-2019            877            John Hancock Demonstration School   
1657   2018-2019            877            John Hancock Demonstration School   
1658   2018-2019            877            John Hancock Demonstration School   
1659   2018-2019            877            John Hancock Demonstration School   
1660   2018-2019            877            John Hancock Demonstration School   
1661 

In [4]:
enrollment_demographics = pd.read_csv('../data/enrollment_demographics.csv')

In [5]:
pd.read_csv('../data/penn_phl_school_ed.csv')

School Year  SRC School ID  \
0    2018-2019            101   
1    2018-2019            102   
2    2018-2019            105   
3    2018-2019            110   
4    2018-2019            113   
5    2018-2019            119   
6    2018-2019            120   
7    2018-2019            123   
8    2018-2019            125   
9    2018-2019            126   
10   2018-2019            128   
11   2018-2019            129   
12   2018-2019            130   
13   2018-2019            134   
14   2018-2019            135   
15   2018-2019            136   
16   2018-2019            137   
17   2018-2019            138   
18   2018-2019            139   
19   2018-2019            141   
20   2018-2019            142   
21   2018-2019            146   
22   2018-2019            147   
23   2018-2019            149   
24   2018-2019            158   
25   2018-2019            200   
26   2018-2019            201   
27   2018-2019            202   
28   2018-2019            205   
29   2018-2019            214   
..         ...            ...   
36   2018-2019            238   
37   2018-2019            239   
38   2018-2019            241   
39   2018-2019            245   
40   2018-2019            247   
41   2018-2019            248   
42   2018-2019            249   
43   2018-2019            251   
44   2018-2019            252   
45   2018-2019            258   
46   2018-2019            259   
47   2018-2019            262   
48   2018-2019            264   
49   2018-2019            265   
50   2018-2019            267   
51   2018-2019            272   
52   2018-2019            403   
53   2018-2019            414   
54   2018-2019            422   
55   2018-2019            427   
56   2018-2019            430   
57   2018-2019            447   
58   2018-2019            453   
59   2018-2019            456   
60   2018-2019            457   
61   2018-2019            508   
62   2018-2019            509   
63   2018-2019            518   
64   2018-2019            856   
65   2018-2019            878   

                                          School Name Learning Network  \
0                            John Bartram High School       Network 13   
1                       West Philadelphia High School     Acceleration   
2         Paul Robeson High School for Human Services        Network 1   
3                        William L. Sayre High School       Network 13   
4                            William T. Tilden School       Network 10   
5                              Motivation High School        Network 1   
6                                   John Barry School     Acceleration   
7                            William C. Bryant School     Acceleration   
8                          Joseph W. Catharine School       Network 12   
9                          Benjamin B. Comegys School       Network 10   
10                             Sadie Alexander School        Network 2   
11                             Andrew Hamilton School        Network 2   
12                         Avery D. Harrington School        Network 2   
13                                Henry C. Lea School        Network 2   
14                       William C. Longstreth School       Network 12   
15                            Morton McMichael School     Acceleration   
16                            S. Weir Mitchell School     Acceleration   
17                            Thomas G. Morton School       Network 12   
18                                Samuel Powel School        Network 2   
19                                James Rhoads School     Acceleration   
20                           Martha Washington School        Network 2   
21                             Add B. Anderson School        Network 2   
22                                 Alain Locke School        Network 2   
23                         Rudolph Blankenburg School     Acceleration   
24                    Middle Years Alternative School        Network 2   
25                     South Phi

In [6]:
penn_schools_ed = pd.read_csv('../data/penn_phl_school_ed.csv')

In [7]:
pd.read_csv('../data/sms_penn_schools.csv')

School  SRC School ID  ULCS Code  \
0                         John Bartram High School            101       1010   
1                    West Philadelphia High School            102       1020   
2      Paul Robeson High School for Human Services            105       1050   
3                     William L. Sayre High School            110       1100   
4                         William T. Tilden School            113       1130   
5                           Motivation High School            119       1190   
6                                John Barry School            120       1200   
7                         William C. Bryant School            123       1230   
8                       Joseph W. Catharine School            125       1250   
9                       Benjamin B. Comegys School            126       1260   
10                          Sadie Alexander School            128       1280   
11                          Andrew Hamilton School            129       1290   
12                      Avery D. Harrington School            130       1300   
13                             Henry C. Lea School            134       1340   
14                    William C. Longstreth School            135       1350   
15                         Morton McMichael School            136       1360   
16                         S. Weir Mitchell School            137       1370   
17                         Thomas G. Morton School            138       1380   
18                             Samuel Powel School            139       1390   
19                             James Rhoads School            141       1410   
20                        Martha Washington School            142       1420   
21                          Add B. Anderson School            146       1460   
22                              Alain Locke School            147       1470   
23                      Rudolph Blankenburg School            149       1490   
24                 Middle Years Alternative School            158       1580   
25                  South Philadelphia High School            200       2000   
26                   Benjamin Franklin High School            201       2010   
27    High School for Creative and Performing Arts            202       2020   
28        Science Leadership Academy Middle School            205       2050   
29                  Julia R. Masterman School H.S.            214       2140   
..                                             ...            ...        ...   
35                 General George A. McCall School            234       2340   
36                      Delaplaine McDaniel School            237       2370   
37                      William M. Meredith School            238       2380   
38                            Robert Morris School            239       2390   
39              Girard Academic Music Program H.S.            241       2410   
40              Girard Academic Music Program M.S.            241       2410   
41                         Edwin M. Stanton School            245       2450   
42                     Albert M. Greenfield School            247       2470   
43                        Chester A. Arthur School            248       2480   
44                          Laura W. Waring School            249       2490   
45                           Andrew Jackson School            251       2510   
46                           Abram S. Jenks School            252       2520   
47                       Eliza B. Kirkbride School            258       2580   
48                       George W. Nebinger School            259       2590   
49                              Academy at Palumbo            262       2620   
50                                Southwark School            264       2640   
51                      Science Leadership Academy            265       2650   
52                        Constitution High School            267       2670   
53                          Vare-Washington 

In [8]:
sms_penn_schools = pd.read_csv('../data/sms_penn_schools.csv')

In [11]:
pd.read_csv('../data/updated_sms_nonpenn_schools.csv')

School SRC School ID  \
0                             John M. Patterson School           140   
1                                  Furness High School           216   
2                                D. Newlin Fell School           219   
3                               F. Amedee Bregy School           224   
4                                Francis S. Key School           254   
5                           George W. Sharswood School           263   
6             The Science Leadership Academy at Beeber           268   
7                               John H. Taggart School           269   
8                                Overbrook High School           402   
9     Murrell Dobbins Career and Technical High School           406   
10              Lewis C. Cassidy Academics Plus School           424   
11                               Samuel Gompers School           428   
12                          Robert E. Lamberton School           432   
13                         E. Washington Rhodes School           435   
14                         Kenderton Elementary School           436   
15                         Overbrook Elementary School           437   
16                             Thomas M. Peirce School           438   
17                              Dr. Ethel Allen School           444   
18                            Tanner G. Duckrey School           446   
19                        Overbrook Educational Center           448   
20                      General George G. Meade School           457   
21                        Thomas A. Edison High School           502   
22                       Philadelphia Military Academy           505   
23   Jules E. Mastbaum Area Vocational Technical Hi...           506   
24                       Parkway Northwest High School           507   
25                       William W. Bodine High School           515   
26                             Penn Treaty High School           516   
27                             Penn Treaty High School           516   
28                              Julia de Burgos School           517   
29                             Alexander Adaire School           520   
..                                                 ...           ...   
224             Sankofa Freedom Academy Charter School           W99   
225           Franklin Towne Charter Elementary School           X03   
226                      Tacony Academy Charter School           X04   
227                      Tacony Academy Charter School           X04   
228                   Aspira Charter School at Stetson           X06   
229                  Mastery Charter School at Harrity           X07   
230                     Mastery Charter School at Mann           X08   
231                  Mastery Charter School at Smedley           X09   
232                Universal Charter School at Bluford           X10   
233                 Universal Charter School at Daroff           X11   
234                 Mastery Charter School at Douglass           X12   
235                     Aspira Charter School at Olney           X14   
236                   Mastery Charter School at Clymer           X15   
237                    Mastery Charter School at Gratz           X16   
238                    Mastery Charter School at Gratz           X16   
239              Universal Charter School at Audenried           X17   
240                   Universal Charter School at Vare           X18   
241                  Lindley Academy Charter at Birney           X19   
242                Mastery Charter School at Cleveland           X20   
243              Universal Charter School at Creighton           X21   
244  The Philadelphia Charter School for Arts and S...           X22   
245  Memphis Street Academy Charter School at J.P. ...           X23   
246                 Universal Charter School at Alcorn           X24   
247                Mastery Charter School at Pastorius           X26   
248             

In [12]:
sms_nonpenn_schools = pd.read_csv('../data/updated_sms_nonpenn_schools.csv')

* Thus far, I have been able to conclude a couple of things from the data I have pulled and analyzed
    - The average number of students enrolled at a  "Penn" school is 467 students while the average number of students enrolled at a "Non Penn" school is 626 students.
    - The average graduation rate at a "Penn" school is 80% but at a "Non Penn" school the average is slightly higher at 83%.
    - The average retention rate of students at a "Penn" school is 83% and at a "Non Penn" school this average is 86%, again slightly higher
    - Looking at attendance, both the "Penn" and "Non Penn" schools had 47% of their students attending more than 95% of instructional days but when it came to percentage of students attending less than 80% of instructional days the average for "Penn" schools was 10% and "Non Penn" schools was 8%

* There isn't much disparity between the two sectors of schools.  Meaning that the proximity of Penn's campus does not have a positive or negative impact on the general standings of each school.  Something I did notice in my analysis however was that there were multiple schools that I flagged because they appeared more than once in my "top 10" lists when looking at retention, graduation, and attendance rates.  I wanted to look to see if they have anything in common, across location and relation to Penn's campus.

* As I continue researching, I continue to develop new sub-sets of research questions.

* I selected these metrics because they were the 5 values that I original looked at across all schools.  Doing this makes it easier to compare all of the flagged schools to each other for more than on value.

* Below are some of the " Penn" schools that I had previously flagged for appearing on more than one list of schools for having the lowest rates in more than one metric ( not necessarily the same metric ).

In [13]:
sms_penn_schools.iloc[[56,3,1,0,25,26],[0,13,257,290,243,255]]

School  Enrollment  Retention Score Grad Score  \
56  Strawberry Mansion High School         293               54         40   
3     William L. Sayre High School         425               67         49   
1    West Philadelphia High School         478               74         66   
0         John Bartram High School         561               71         67   
25  South Philadelphia High School         559               78         53   
26   Benjamin Franklin High School         506               67         59   

    Attendance (95%+) Score  Attendance (<80%) Score  
56                       14                       49  
3                        14                       49  
1                        14                       40  
0                        26                       35  
25                       32                       27  
26                       17                       38

* Looking at the given data above, it's important to mark the similarities and the differences between all of the schools:
    - School enrollment for example, is not similar across the schools with the highest being 561 and the lowest being 293
    - The graduation rate also had some range; the highest percentage being 67% and the lowest being 40%.
    - However, there is a lot of similarity across metrics between Strawberry Mansion High School and William L. Sayre High School.  I'm interested if these schools have any other similarities across additional score card values.


* Below are some of the "Penn" schools that I had previously flagged for appearing on more than one list of schools for having the highest rate in more than one metric.

In [14]:
sms_penn_schools.iloc[[29,54,49,39,46,10],[0,13,257,290,243,255]]

School  Enrollment  Retention Score  \
29               Julia R. Masterman School H.S.         451               99   
54  High School of Engineering and Science H.S.         777               98   
49                           Academy at Palumbo         924               97   
39           Girard Academic Music Program H.S.         253               98   
46                        Abram S. Jenks School         300               84   
10                       Sadie Alexander School         579               92   

   Grad Score  Attendance (95%+) Score  Attendance (<80%) Score  
29         99                       92                        0  
54         98                       77                        2  
49         96                       74                        3  
39        100                       63                        2  
46          .                       77                        1  
10          .                       80                        1

* Being middle schools, both the Abram S Jenks School and the Sadie Alexander School do not have graduation rates.

* Looking at the given data above, it's important to mark the similarities and the difference between all of the schools:
    - School enrollment is again not similar across this spectrum schools with 924 being the highest amount of students enrolled and 253 being the lowest.
    -  Across Retention Score, Graduation Score, and Score of Attendance (<80%) the schools are almost identical.
    - There is a big range of school scores with Score of Attendance (95%+).
    - The most similar schools here score wise are Julia R. Masterman High School and Sadie Alexander School.  However, they differ in the fact that one is a high school and one is a middle school.
    - Personally I see Julia R. Masterman School as having performed the best across all metrics.

* Below are some of the "Non Penn" schools that I had previously flagged for appearing on more than one list of schools for having the lowest rate in more than one metric.

In [15]:
sms_nonpenn_schools.iloc[[8,64,89,235,59,21,78],[0,13,257,290,243,255]]

School  Enrollment  Retention Score Grad Score  \
8             Overbrook High School         568               64         50   
64   Martin Luther King High School         667               65         48   
89            Frankford High School         955               78         55   
235  Aspira Charter School at Olney        2006               78         71   
59           Kensington High School         579               75         63   
21     Thomas A. Edison High School        1055               69         56   
78        Theodore Roosevelt School         525               58          .   

     Attendance (95%+) Score  Attendance (<80%) Score  
8                         16                       48  
64                        16                       41  
89                        21                       34  
235                        7                       44  
59                        25                       37  
21                        19                       36  
78                        33                       24

* Looking at the given data above, it's important to mark the similarities and the differences between all of the schools:
    - There is no similarity across student enrollment for any of the schools.
    - Retention rates and graduation rates, again present a range of scores; with the lowest retention rates being 58% and the highest being 78% and the lowest graduation rate being 48% and the highest being 71%.
    - Looking at the attendance scores for both (95%+) and (<80%) there is again a range of scores 
    - There is a similarity between the Overbook High School and Martin Luther King High School and a similarity between Frankford High School and Thomas A. Edison High School that should be explored further across different metrics.
    
* My goal is to see if there is try and determine if there is any causal reason(s) why certain schools have the low graduation and/or attendance and/or retention scores.

* Below are some of the "Non Penn" schools that I had previously flagged for appearing on more than one list of schools for having the highest rate in more than one metric.

In [16]:
sms_nonpenn_schools.iloc[[60,172,88,56,25,158,29],[0,13,257,290,243,255]]

School  Enrollment  \
60                                 Central High School        2386   
172  Mathematics, Science, and Technology Community...         427   
88                                Lankenau High School         348   
56                                        The U School         291   
25                       William W. Bodine High School         515   
158               Multicultural Academy Charter School         279   
29                             Alexander Adaire School         356   

     Retention Score Grad Score  Attendance (95%+) Score  \
60                99        100                       85   
172               99         97                       63   
88                94        100                       53   
56                79         88                       95   
25                95         95                       64   
158               85         95                       50   
29                94          .                       68   

     Attendance (<80%) Score  
60                         1  
172                        0  
88                         6  
56                         1  
25                         2  
158                        4  
29                         0

* Looking at the given data above, it's important to mark the similarities and the difference between all of the schools:
    - The enrollment for all of the schools is relatively similar except for Central High School which has 2386 students enrolled.
    - The retention score of each of the schools is in the 90 percentile except for the U School and for Alexander Adaire School.
    - Looking at the attendance scores there are also a few outliers. The Multicultural Academy Charter and Lankenau High School both have low scores for 95%+ attendance and high scores for <80% attendance, but have almost perfect graduation and retention rates.
    - Personally I see Central High School as having performed the best across all of the metrics.

A similar theme I have seen across the "Penn" and "Non Penn" schools that have some high metric scores is that there does not seem to be a theme at all.  Not many schools have performed similarly across all of the metrics, but will instead have one high performing value.  So I am not able to conclude a causal relation between metrics.  The only causal relation that I can determine at this point is between the attendance rates.  A high "Attendance (95%+) Score" typically indicates a low "Attendance (<80%) Score" and vise versa.